### PR TITLE
Add TLSConfig to RedisConfig and NewRedisClient #patch

### PR DESCRIPTION
**Pull request by bimtauer**
# TL;DR
Adds the TLSConfig field to respective structs to allow for secure Redis communication. 
Work in progress.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Based on the entirety of possible [redis options](https://github.com/go-redis/redis/blob/ce40cd942a72c4a93f9025047e6fd3f510700ab3/universal.go), I added the TLSConfig field to the instantiation in NewRedisClient and also extended flytes own RedisConfig with that field. 

## Tracking Issue
No issue yet, do you want me to open one?

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/controller/nodes/task/resourcemanager/config/config.go [hunk 1]
```diff
@@ -38,9 +41,53 @@ type RedisConfig struct {
 	HostPaths   []string `json:"hostPaths" pflag:",Redis hosts locations."`
 	PrimaryName string   `json:"primaryName" pflag:",Redis primary name, fill in only if you are connecting to a redis sentinel cluster."`
 	// deprecated: Please use HostPaths instead
-	HostPath   string `json:"hostPath" pflag:",Redis host location"`
-	HostKey    string `json:"hostKey" pflag:",Key for local Redis access"`
-	MaxRetries int    `json:"maxRetries" pflag:",See Redis client options for more info"`
+	HostPath   string    `json:"hostPath" pflag:",Redis host location"`
+	HostKey    string    `json:"hostKey" pflag:",Key for local Redis access"`
+	MaxRetries int       `json:"maxRetries" pflag:",See Redis client options for more info"`
+	TLSConfig  TLSConfig `json:"tlsConfig" pflag:",See the crytpo/tls config object for more info"`
```
Comment (EngHabu) [inline pkg/controller/nodes/task/resourcemanager/config/config.go:47]:
```suggestion
	TLS  TLSConfig `json:"tls" pflag:",See the crytpo/tls config object for more info"`
```

File: pkg/controller/nodes/task/resourcemanager/config/config.go [hunk 2]
```diff
@@ -38,9 +41,53 @@ type RedisConfig struct {
...
+
+func ParseTLSConfig(cfg TLSConfig) *tls.Config {
+	// Parses TLSConfig settings into a proper tls.Config object. Returns a pointer
+	// to nil if no settings were manually defined (i.e. cfg is empty) to keep tls disabled.
+	if cfg != (TLSConfig{}) {
+		/* #nosec G402 */
+		return &tls.Config{
+			//Certificates []Certificate,
+			//RootCAs *x509.CertPool,
+			ServerName: cfg.ServerName,
```
Comment (EngHabu) [inline pkg/controller/nodes/task/resourcemanager/config/config.go:84]: Is it enough to configure server name and not CA & Certs? is this just to work around common name mismatch issues?
